owasp-top10-2017-apps/a1/copy-n-paste/app/util/db.go

```diff
@@ -108,12 +117,16 @@ func CheckIfUserExists(username string) (bool, error) {
 	}
 	defer dbConn.Close()
 
-	query := fmt.Sprint("select username from Users where username = '" + username + "'")
-	rows, err := dbConn.Query(query)
+	Stmt, err := dbConn.Prepare("select username from Users where username = ?")
 	if err != nil {
-		return false, err
+		log.Fatal(err)
+	}
+	defer Stmt.Close()
+
+	rows, err := Stmt.Query(username)
+	if err != nil {
+		log.Fatal(err)
 	}
-	defer rows.Close()
```

Comment:
I think a `defer rows.Close()` is missing in this function as well. Would you mind confirming this also, please? 😄